src/controller/java/AndroidCallbacks.cpp

```diff
@@ -229,18 +202,17 @@ void ReportCallback::OnReportEnd()
     CHIP_ERROR err = CHIP_NO_ERROR;
     JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
     VerifyOrReturn(env != nullptr, ChipLogError(Controller, "Could not get JNIEnv for current thread"));
+
     JniLocalReferenceScope scope(env);
-    VerifyOrReturn(mReportCallbackRef.HasValidObjectRef(),
-                   ChipLogError(Controller, "mReportCallbackRef is not valid in %s", __func__));
-    jobject reportCallback = mReportCallbackRef.ObjectRef();
-    JniGlobalReference globalRef(std::move(mNodeStateObj));
-    jmethodID onReportMethod;
-    err = JniReferences::GetInstance().FindMethod(env, reportCallback, "onReport", "(Lchip/devicecontroller/model/NodeState;)V",
-                                                  &onReportMethod);
-    VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find onReport method"));
+    VerifyOrReturn(mWrapperCallbackRef.HasValidObjectRef(),
+                   ChipLogError(Controller, "mWrapperCallbackRef is not valid in %s", __func__));
+    jobject wrapperCallback = mWrapperCallbackRef.ObjectRef();
```

Comment:
where do you release nodeStateObject, existing logic would pass nodeState to application via OnReport, and further release nodeState logic when we call OnReportEnd.